src/diffusers/pipelines/pyramid_broadcast_utils.py

```diff
@@ -120,40 +32,68 @@ def _enable_pyramid_attention_broadcast(self) -> None:
 
         for name, module in denoiser.named_modules():
             if isinstance(module, Attention):
-                logger.debug(f"Enabling Pyramid Attention Broadcast in layer: {name}")
+                is_spatial_attention = (
+                    any(x in name for x in self._pab_spatial_attn_layer_identifiers)
+                    and self._pab_spatial_attn_skip_range is not None
+                    and not module.is_cross_attention
+                )
+                is_temporal_attention = (
+                    any(x in name for x in self._pab_temporal_attn_layer_identifiers)
+                    and self._pab_temporal_attn_skip_range is not None
+                    and not module.is_cross_attention
+                )
+                is_cross_attention = (
+                    any(x in name for x in self._pab_cross_attn_layer_identifiers)
+                    and self._pab_cross_attn_skip_range is not None
+                    and module.is_cross_attention
+                )
```

Comment:
Should there be a place for joint attention / SD3 style attention here?

Reply:
Nope. Those are equivalent to spatial blocks as described in the paper. Skipping too much of their computation will result in bad generations, so the configuration for spatial attention would apply here (skip every 2nd iteration through the attention processor between timestep 100 to 800 by default recommendations)

A slightly more detailed explanation here: https://github.com/huggingface/diffusers/pull/9562#discussion_r1807502622